dds/DCPS/TypeSupportImpl.cpp

```diff
@@ -84,7 +84,11 @@ void TypeSupportImpl::populate_dependencies(const RcHandle<XTypes::TypeLookupSer
       const size_t tobj_size = serialized_size(XTypes::get_typeobject_encoding(), iter->second);
       XTypes::TypeIdentifierWithSize tmp = {*it, static_cast<ACE_CDR::ULong>(tobj_size)};
       deps_with_size.append(tmp);
-    } else if (DCPS_debug_level >= 1) {
+    } else if ((DCPS_debug_level >= 1) &&
+               (it->kind() ==  XTypes::TI_STRONGLY_CONNECTED_COMPONENT ||
+                it->kind() == XTypes::EK_COMPLETE ||
+                it->kind() == XTypes::EK_MINIMAL ||
+                it->kind() == XTypes::TK_ANNOTATION)) {
```

Comment:
```suggestion
    } else if ((DCPS_debug_level >= 1) && (it->kind() == XTypes::EK_MINIMAL)) {
```
This should be sufficient because `it` is a minimal TypeIdentifier, and `TK_ANNOTATION` isn't a TypeIdentifier kind and `TI_STRONGLY_CONNECTED_COMPONENT` doesn't have TypeObject.